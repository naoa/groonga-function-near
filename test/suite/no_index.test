register functions/near
table_create Diaries TABLE_NO_KEY
column_create Diaries title COLUMN_SCALAR ShortText
column_create Diaries body COLUMN_SCALAR ShortText

load --table Diaries
[
  {
    "title": "2014-8-12's weather",
    "body": "It was rainy in Tokyo. But it was sunny in Osaka."
  },
  {
    "title": "2014-8-13's weather",
    "body": "It was cloudy in Tokyo. But it was rainy in Sapporo."
  },
  {
    "title": "2014-8-15's weather",
    "body": "It was sunny in Osaka. But it was snowy in Tokyo."
  }
]

select Diaries \
  --output_columns body,_score \
  --query '_id:3' \
  --filter 'near(body, 2, "rainy tokyo")'
