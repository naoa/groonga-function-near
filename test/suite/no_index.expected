register functions/near
[[0,0.0,0.0],true]
table_create Diaries TABLE_NO_KEY
[[0,0.0,0.0],true]
column_create Diaries title COLUMN_SCALAR ShortText
[[0,0.0,0.0],true]
column_create Diaries body COLUMN_SCALAR ShortText
[[0,0.0,0.0],true]
load --table Diaries
[
  {
    "title": "2014-8-12's weather",
    "body": "It was rainy in Tokyo. But it was sunny in Osaka."
  },
  {
    "title": "2014-8-13's weather",
    "body": "It was cloudy in Tokyo. But it was rainy in Sapporo."
  },
  {
    "title": "2014-8-15's weather",
    "body": "It was sunny in Osaka. But it was snowy in Tokyo."
  }
]
[[0,0.0,0.0],3]
select Diaries   --output_columns body,_score   --query '_id:3'   --filter 'near(body, 2, "rainy tokyo")'
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        1
      ],
      [
        [
          "body",
          "ShortText"
        ],
        [
          "_score",
          "Int32"
        ]
      ],
      [
        "It was sunny in Osaka. But it was snowy in Tokyo.",
        2
      ]
    ]
  ]
]
